src/thor/bidirectional_astar.cc

```diff
@@ -622,6 +622,8 @@ BidirectionalAStar::GetBestPath(valhalla::Location& origin,
   BDEdgeLabel fwd_pred, rev_pred;
   bool expand_forward = true;
   bool expand_reverse = true;
+  bool forward_exhausted = false;
+  bool reverse_exhausted = false;
```

Comment:
why do we need these additional variables ? we can check if `forward_pred_idx` equals to `kInvalidLabel` or not

Reply:
We could (i initially used that in one of my initial iterations), but found the bools a bit more readable and make the intent more explicit.

Reply:
Perhaps a best of both worlds could be `bool is_forward_exhausted() { return "forward_pred_idx == kInvalidLabel" }`?